fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/EraseResourceDAO.java

```diff
@@ -0,0 +1,351 @@
+ */
+public class EraseResourceDAO extends ResourceDAOImpl {
+
+    private static final String CLASSNAME = EraseResourceDAO.class.getSimpleName();
+    private static final Logger LOG = Logger.getLogger(CLASSNAME);
+
+    private static final String CALL_POSTGRES = "{CALL %s.ERASE_RESOURCE(?, ?, ?)}";
+    private static final String CALL_DB2 = "CALL %s.ERASE_RESOURCE(?, ?, ?)";
+
+    // The translator specific to the database type we're working with
+    private final IDatabaseTranslator translator;
+
+    private ResourceEraseRecord eraseRecord;
+    private EraseDTO eraseDto;
+    private Connection conn;
+
+    /**
+     * Public constructor
+     *
+     * @param connection
+     * @param translator
+     * @param schemaName
+     * @param flavor
+     * @param cache
+     * @param rrd
+     */
+    public EraseResourceDAO(Connection conn, IDatabaseTranslator translator, String schemaName, FHIRDbFlavor flavor, FHIRPersistenceJDBCCache cache,
+            IResourceReferenceDAO rrd) {
+        super(conn, schemaName, flavor, cache, rrd);
+        this.translator = translator;
+        this.conn = getConnection();
+    }
+
+    /**
+     * Execute the stored procedure/function to erase the content.
+     *
+     * @param callStr the sql that should be executed
+     * @throws Exception
+     */
+    private void runCallableStatement(String callStr) throws Exception {
+        try (CallableStatement call = conn.prepareCall(String.format(callStr, getSchemaName()));){
+            call.setString(1, eraseDto.getResourceType());
+            call.setString(2, eraseDto.getLogicalId());
+            call.registerOutParameter(3, Types.BIGINT);
+            call.execute();
+
+            Integer deleted = (int) call.getLong(3);
+            if (LOG.isLoggable(Level.FINEST)) {
+                LOG.finest("Deleted from [" + eraseDto.getResourceType() + "/" + eraseDto.getLogicalId() + "] deleted [" + deleted + "]");
+            }
+
+            eraseRecord.setTotal(deleted);
+
+            ResourceEraseRecord.Status status = ResourceEraseRecord.Status.DONE;
+            if (deleted == -1) {
+                status = ResourceEraseRecord.Status.NOT_FOUND;
+            }
+            eraseRecord.setStatus(status);
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, "SQL Call to Erase Failed to execute", x);
+            throw translator.translate(x);
+        }
+    }
+
+    /**
+     * Executes the SQL logic as part of the dao rather than via a stored procedure/function.
+     *
+     * @throws SQLException
+     */
+    public void runInDao() throws SQLException {
+        String resourceType = eraseDto.getResourceType();
+        String logicalId = eraseDto.getLogicalId();
+
+        long resourceTypeId = -1;
+        long logicalResourceId = -1;
+        long resourceId = -1;
+        int version = -1;
+        Integer total = 0;
+
+        // Prep 1: Get the v_resource_type_id
+        final String GET_RESOURCE_TYPE_ID =
```

Comment:
resource_type_id can be obtained directly from the FHIRPersistenceJDBCCache cache member. We preload these because they are static and required so often.

Reply:
I've changed, and now it uses the cache in the Unit Tests as well.